warp10/src/main/java/io/warp10/script/functions/TIMEBOX.java

```diff
@@ -44,30 +44,82 @@
   /**
    * Maximum timeboxing possible
    */
-  private static final long TIMEBOX_MAXTIME;
-  
-  static {
-    TIMEBOX_MAXTIME = Long.parseLong(WarpConfig.getProperty(Configuration.CONFIG_WARPSCRIPT_TIMEBOX_MAXTIME, Long.toString(DEFAULT_TIMEBOX_MAXTIME)));
+  private final long TIMEBOX_MAXTIME;
+
+  /**
+   * Allowance capability to raise TIMEBOX_MAXTIME
+   */
+  private final String SPECIAL_ALLOWANCE_CAPNAME;
+
+  /**
+   * Wrapped function
+   */
+  private final WarpScriptStackFunction function;
+
+  public TIMEBOX(String name, WarpScriptStackFunction function) {
+    this(name, function, false);
   }
-  
-  public TIMEBOX(String name) {
+
+  public TIMEBOX(String name, WarpScriptStackFunction function, Boolean requireCapability) {
```

Comment:
This is not needed, functions can simply be included in a macro.

Reply:
but macros don't have a name and must share the same maxtime

Reply:
There is no need to allow setting the maxtime per function. The wrapped functions would also not be snapshotable otherwise the snapshot would allow the setting of the various parameters which would expose to security risks.

Reply:
And also this simplifies registration. For example, suppose an extension wants to timebox the function DBSCAN, here we could simply do:
`functions.add("DBSCAN", TIMEBOX(new DBSCAN("DBSCAN"), "extension.maxtime.capname")`
rather than building the macro with capability check for each function

Reply:
As I've said this would require to allow the explicit setting of sensitive parameters, so this is not a feature that should be included in the core of Warp 10. It may be in an extension that users decide to explicitly allow, but not in the core.

Reply:
I don't see why it can't be snapshoted, the snapshot would just be the name of the wrapped function. The fact that maxtime limits would be different on other instances, or that the capability that control it is different is not an issue.